driver/src/solution_submission/mod.rs

```diff
@@ -92,8 +103,15 @@ impl<'a> StableXSolutionSubmitting for StableXSolutionSubmitter<'a> {
         batch_index: U256,
         solution: Solution,
     ) -> Result<U256, SolutionSubmissionError> {
+        // NOTE: Compare with `>=` as the exchange's current batch index is the
+        //   one accepting orders and does not yet accept solutions.
+        while batch_index.as_u32() >= self.contract.get_current_auction_index()? {
```

Comment:
Why don't we have to `SolutionSubmissionError::from` here, but below?

Reply:
`?` implicitly calls `From::from` for error conversion. The line below is for mapping the `Result<_, E>` to `Result<_, SolutionSumissionError>` - it could be re-written (and is equivalent to):
```
Ok(self.contract.get_solution_objective_value(batch_index, solution, None)?)
```

Reply:
Thanks!

👨‍🏫